src/VecSim/types/float16.h

```diff
@@ -10,8 +10,12 @@
 #include <cstring>
 #include <algorithm>
 namespace vecsim_types {
-
-using float16 = uint16_t;
+struct float16 {
+    uint16_t val;
+    constexpr float16(uint16_t val = 0) : val(val) {}
```

Comment:
is it beneficial to have an "undefined" variant for declaration only? (like `int x;`)

Reply:
Since 0 is the default value It will be initialized to 0